### ISSUE TITLE
bem-forms доопределение поведения в JavaScript

### ISSUE THREAD

**Issue by bradbenetton** [bem/bem-forum-content-ru#384]
Всем привет!

Начинаю изучать BEM, пытаюсь поработать с библотекой [bem-forms](https://github.com/bem/bem-forms). Библиотеку подключил, прописал путь до библиотеки в `.bem/make.js`. `BEMHTML` библиотеки отрабатывает, html-tag формы преобразуется из `div` в `form`. Не получается доопределить поведение формы в `JavaScript`. Пытаюсь сделать это следующем образом, как указано [здесь](https://ru.bem.info/technology/i-bem/v2/i-bem-js/#Доопределение-существующего-блока).
```js
modules.define('form', function(provide, Form) {
    provide(Form.decl({
        onSetMod: {
            'js': {
                'inited': function() {
                    console.log('form started ...');
                }
            }
        }
    }));
});
```

При этом поведение блока `button` из библиотеки `bem-components` таким же образом переопределить получается.
Использую `bem-tools` и `project-stub`.

Пока писал, заметил, что у формы не проставляется модификатор `js_inited`. Как можно поправить? В `BEMJSON` явно указал `js: true`, но ничего не происходит.

P.S. И не большой вопрос, при создании нового блока, bem-tools создает файл `blockName.browser.js`. Что значит `browser.js`? Чтобы указать, что в файле клиентский код?

Заранее спасибо!

**Comment by zxqfox** [comment]
> modules.define('form', function(provide, Form) {

Возможно, тут не хватает зависимости от `['i-bem__dom']`,
Еще, похоже, забыли про поддержку bem-tools ;-( https://github.com/bem/bem-forms/tree/master/common.blocks — не вижу .bem папок на уровнях

**Comment by tadatuta** [comment]
@bradbenetton 
>  Что значит browser.js? Чтобы указать, что в файле клиентский код?

Да, т.к. поддерживается еще `node.js` для сервера. Кроме `browser.js` в сборку клиентского кода попадут также файлы просто с `js`, их следует воспринимать как синонимы.

**Comment by zxqfox** [comment]
@bradbenetton Кстати, это тоже может быть проблемой и причиной отсутствия инициализации. `enb-diverse-js` подключается? `?.browser.js` входит в сборку на клиент?

**Comment by adinvadim** [comment]
@zxqfox  у меня такая же проблеам возникла, использую bem-forms#v1.0.0 , enb, bemdom в зависимости добавил, а inited не ставится(

```
modules.define(
    'form',
    ['i-bem__dom'],
    function(provide, Form, BEMDOM) {

Form.decl(this.name, {

    onSetMod: {
        'js': {
            'inited': function() {
                this.__base.apply(this, arguments);
                console.log(this.domElem)
            }
        }
    }

});

provide(Form)
})
```

**Comment by zxqfox** [comment]
Нужен еще кусок bemjson с формой.

js: true есть?

**Comment by adinvadim** [comment]
https://github.com/bem-incubator/bem-forms/blob/v1.0.0/common.blocks/form/form.bemhtml#L5
да и в bemjson есть, на всех формах, даже которые я не переоредлял нет js_inited. 

Я понял, это наверное из-за линивой иницилазции :)

**Comment by zxqfox** [comment]
А, точно. ;-)

**Comment by awinogradov** [comment]
Примеры доопределения есть тут https://github.com/bem-incubator/bem-forms/tree/v1.0.0/example.blocks , а вот bem-tools починим да)
